packages/boundless-popover/index.js

```diff
@@ -411,16 +417,34 @@ export default class Popover extends React.PureComponent {
         this.$caret.style[longitudinal ? 'top' : 'left'] = '0px';
     }
 
+    matchAnchorWidth = () => {
+      if (!this.props.matchAnchorWidth) return;
+
+      const anchor = this.props.anchor instanceof HTMLElement
+                     ? this.props.anchor
+                     : findDOMNode(this.props.anchor);
+      const dialog = this.dialog.$dialog;
+      const anchorWidth = anchor.getBoundingClientRect().width;
+      dialog.style.width = `${anchorWidth}px`;
+    }
+
     componentDidMount() {
+        this.matchAnchorWidth();
         this.align();
+        window.addEventListener('resize', this.matchAnchorWidth, true);
```

Comment:
I think we can remove this additional event listener as well by integrating the width matching function into the alignment flow.